### ISSUE TITLE
Hubspot integration mapping to array does not implode with ';'

### ISSUE THREAD

**Issue by jodeba** [verbb/formie#1272]
### Describe the bug

I mapped checkboxes to a array type field in Hubspot 

This is the object in the integration cache:
```{
        "class": "verbb\\formie\\models\\IntegrationField",
        "handle": "sector__newsletter_",
        "name": "Sector (newsletter) ",
        "type": "array",
        "required": null,
        "options": {
          "label": "Sector (newsletter) ",
          "options": [
           ...
          ]
        }
      },
```

But they still are send as a string "value 1, value2", in stead of "value 1;value 2"

### Steps to reproduce

1. Create checkboxes field
2. Map to Hubspot checkboxes field
3. Send submission

### Form settings

- Multi-page form: No
- Submission Method: both
- Client-side Validation: No
- Custom Form Templates: Yes

### Craft CMS version

Craft Pro 4.3.6.1

### Plugin version

2.0.21

### Multi-site?

Yes

### Additional context

I tried to follow the code:

At [this point](https://github.com/verbb/formie/blob/206b9e712522bba7dfc79b21c07007059aa1e9ff/src/base/Crm.php#L72) the fields are correctly an array of IntegrationField, but in [Integration:565](https://github.com/verbb/formie/blob/206b9e712522bba7dfc79b21c07007059aa1e9ff/src/base/Integration.php#L565) they are a array of collections, so the tag cannot be found.

**Comment by engram-design** [comment]
Can you let me know what type of field you're mapping to in Hubspot? Just wanted to confirm it was a Checkboxes field, as that's exactly what I'm mapping to with everything working.

<img width="558" alt="image" src="https://user-images.githubusercontent.com/1221575/216459081-c5b4812e-121f-4ef1-8d47-e9c905714a40.png">

The check [here](https://github.com/verbb/formie/blob/206b9e712522bba7dfc79b21c07007059aa1e9ff/src/integrations/crm/HubSpot.php#L82) sorts out the custom delimiter, and does work for me on a few array-based fields.

You can see this is importing correctly in a "Checkboxes" field in HubSpot

<img width="412" alt="image" src="https://user-images.githubusercontent.com/1221575/216460397-3d82d49e-fcc0-4a43-b78f-b320f8e51d4c.png">

**Comment by jodeba** [comment]
Hi

the Hubspot field is of type "Multiple checkboxes",

the integrationField->getType() is indeed 'array', but the code doesn't reach that point because [$fieldSettings here](https://github.com/verbb/formie/blob/206b9e712522bba7dfc79b21c07007059aa1e9ff/src/base/Integration.php#L565) is not an array.

**Comment by engram-design** [comment]
Thanks for confirming. I wasn't sure what you meant by:

> but in [Integration:565](https://github.com/verbb/formie/blob/206b9e712522bba7dfc79b21c07007059aa1e9ff/src/base/Integration.php#L565) they are a array of collections

What's the collection you're referring to? Just a quick example of adding `Craft::dd($fieldSettings);` at [this line](https://github.com/verbb/formie/blob/206b9e712522bba7dfc79b21c07007059aa1e9ff/src/base/Integration.php#L563) produces an array of `IntegrationField` objects, which is correct.

```
[
    0 => verbb\formie\models\IntegrationField#1
    (
        [handle] => 'email'
        [name] => 'Email'
        [type] => null
        [required] => '1'
        [options] => []
        [yii\base\Model:_errors] => null
        [yii\base\Model:_validators] => null
        [yii\base\Model:_scenario] => 'default'
        [yii\base\Component:_events] => []
        [yii\base\Component:_eventWildcards] => []
        [yii\base\Component:_behaviors] => []
    )
    1 => verbb\formie\models\IntegrationField#2
    (
        [handle] => 'checkboxes'
        [name] => 'Checkboxes'
        [type] => 'array'
        [required] => null
        [options] => [
            'label' => 'Checkboxes'
            'options' => [
                0 => [
                    'label' => 'Option 1'
                    'value' => 'Option 1'
                ]
                1 => [
                    'label' => 'Option 2'
                    'value' => 'Option 2'
                ]
                2 => [
                    'label' => 'Option 3'
                    'value' => 'Option 3'
                ]
            ]
        ]
        [yii\base\Model:_errors] => null
        [yii\base\Model:_validators] => null
        [yii\base\Model:_scenario] => 'default'
        [yii\base\Component:_events] => []
        [yii\base\Component:_eventWildcards] => []
        [yii\base\Component:_behaviors] => []
    )
```

Can you let me know what you see on your end?

**Comment by jodeba** [comment]
This is my output at that point:

```
array:5 [
  0 => verbb\formie\models\IntegrationCollection {#1554
    -_events: []
    -_eventWildcards: []
    -_behaviors: []
    -_errors: null
    -_validators: null
    -_scenario: "default"
    +id: "7894001__2a2caac2-03b3-4571-9104-56ae6da3821b"
    +name: "2022 Solliciteer nu, en bezorg ons je CV"
    +fields: array:7 [
      0 => verbb\formie\models\IntegrationField {#1468
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "trackingID"
        +name: "Tracking ID"
        +type: null
        +required: null
        +options: []
      }
      1 => verbb\formie\models\IntegrationField {#1467
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "firstname"
        +name: "Voornaam"
        +type: "string"
        +required: null
        +options: []
      }
      2 => verbb\formie\models\IntegrationField {#1466
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "lastname"
        +name: "Naam"
        +type: "string"
        +required: null
        +options: []
      }
      3 => verbb\formie\models\IntegrationField {#1465
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "email"
        +name: "E-mail"
        +type: "string"
        +required: null
        +options: []
      }
      4 => verbb\formie\models\IntegrationField {#1464
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "vraag_of_motivatie"
        +name: "Vraag of motivatie "
        +type: "string"
        +required: null
        +options: []
      }
      5 => verbb\formie\models\IntegrationField {#1463
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "jobtitle"
        +name: "Functietitel"
        +type: "string"
        +required: null
        +options: []
      }
      6 => verbb\formie\models\IntegrationField {#1462
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "sector"
        +name: "In welke sector ben je op zoek?"
        +type: "array"
        +required: null
        +options: array:2 [
          "label" => "In welke sector ben je op zoek?"
          "options" => array:8 [
            0 => array:2 [
              "label" => "Engineering"
              "value" => "Engineering"
            ]
            1 => array:2 [
              "label" => "IT Analytical"
              "value" => "IT Analytical"
            ]
            2 => array:2 [
              "label" => "IT Freelance"
              "value" => "IT Freelance"
            ]
            3 => array:2 [
              "label" => "IT Leadership"
              "value" => "IT Leadership"
            ]
            4 => array:2 [
              "label" => "Madison Intern"
              "value" => "Madison Intern"
            ]
            5 => array:2 [
              "label" => "Software"
              "value" => "Software"
            ]
            6 => array:2 [
              "label" => "Supply Chain "
              "value" => "Supply Chain"
            ]
            7 => array:2 [
              "label" => "Techniekers"
              "value" => "Techniekers"
            ]
          ]
        ]
      }
    ]
  }
  1 => verbb\formie\models\IntegrationCollection {#1548
    -_events: []
    -_eventWildcards: []
    -_behaviors: []
    -_errors: null
    -_validators: null
    -_scenario: "default"
    +id: "7894001__4de20340-d2c2-4770-8560-2cfd36635228"
    +name: "2023 Nieuwsbrief Formulier"
    +fields: array:5 [
      0 => verbb\formie\models\IntegrationField {#1553
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "trackingID"
        +name: "Tracking ID"
        +type: null
        +required: null
        +options: []
      }
      1 => verbb\formie\models\IntegrationField {#1552
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "legalConsentOptions"
        +name: "Legal Consent Options"
        +type: null
        +required: null
        +options: []
      }
      2 => verbb\formie\models\IntegrationField {#1551
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "sector__newsletter_"
        +name: "Van welke sector wil je een nieuwsbrief ontvangen?"
        +type: "array"
        +required: null
        +options: array:2 [
          "label" => "Van welke sector wil je een nieuwsbrief ontvangen?"
          "options" => array:8 [
            0 => array:2 [
              "label" => "Engineering "
              "value" => "Engineering"
            ]
            1 => array:2 [
              "label" => "IT Analytical "
              "value" => "IT Analytical"
            ]
            2 => array:2 [
              "label" => "IT Freelance"
              "value" => "IT Freelance"
            ]
            3 => array:2 [
              "label" => "IT Leadership"
              "value" => "IT Leadership"
            ]
            4 => array:2 [
              "label" => "Madison Intern"
              "value" => "Madison Intern"
            ]
            5 => array:2 [
              "label" => "Software"
              "value" => "Software"
            ]
            6 => array:2 [
              "label" => "Supply Chain "
              "value" => "Supply Chain"
            ]
            7 => array:2 [
              "label" => "Techniekers "
              "value" => "Techniekers"
            ]
          ]
        ]
      }
      3 => verbb\formie\models\IntegrationField {#1550
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "email"
        +name: "E-mail"
        +type: "string"
        +required: null
        +options: []
      }
      4 => verbb\formie\models\IntegrationField {#1549
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "cities_of_belgium"
        +name: "Locatie"
        +type: "string"
        +required: null
        +options: []
      }
    ]
  }
  2 => verbb\formie\models\IntegrationCollection {#1540
    -_events: []
    -_eventWildcards: []
    -_behaviors: []
    -_errors: null
    -_validators: null
    -_scenario: "default"
    +id: "7894001__678f4717-47ff-4d65-96a4-fe1d0c5021e9"
    +name: "2022 Zet ons aan het werk "
    +fields: array:7 [
      0 => verbb\formie\models\IntegrationField {#1547
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "trackingID"
        +name: "Tracking ID"
        +type: null
        +required: null
        +options: []
      }
      1 => verbb\formie\models\IntegrationField {#1546
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "voorkeur_werklocatie"
        +name: "Waar wil je graag werken?"
        +type: "string"
        +required: null
        +options: array:2 [
          "label" => "Waar wil je graag werken?"
          "options" => array:2 [
            0 => array:2 [
              "label" => "Ik wil graag in de buurt werken"
              "value" => "Ik wil graag in de buurt werken"
            ]
            1 => array:2 [
              "label" => "Maakt niet uit, alles is goed"
              "value" => "Maakt niet uit, alles is goed"
            ]
          ]
        ]
      }
      2 => verbb\formie\models\IntegrationField {#1545
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "sector"
        +name: "In welke sector ben je op zoek?"
        +type: "array"
        +required: null
        +options: array:2 [
          "label" => "In welke sector ben je op zoek?"
          "options" => array:8 [
            0 => array:2 [
              "label" => "Engineering"
              "value" => "Engineering"
            ]
            1 => array:2 [
              "label" => "IT Analytical"
              "value" => "IT Analytical"
            ]
            2 => array:2 [
              "label" => "IT Freelance"
              "value" => "IT Freelance"
            ]
            3 => array:2 [
              "label" => "IT Leadership"
              "value" => "IT Leadership"
            ]
            4 => array:2 [
              "label" => "Madison Intern"
              "value" => "Madison Intern"
            ]
            5 => array:2 [
              "label" => "Software"
              "value" => "Software"
            ]
            6 => array:2 [
              "label" => "Supply Chain "
              "value" => "Supply Chain"
            ]
            7 => array:2 [
              "label" => "Techniekers"
              "value" => "Techniekers"
            ]
          ]
        ]
      }
      3 => verbb\formie\models\IntegrationField {#1544
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "firstname"
        +name: "Voornaam"
        +type: "string"
        +required: null
        +options: []
      }
      4 => verbb\formie\models\IntegrationField {#1543
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "lastname"
        +name: "Naam"
        +type: "string"
        +required: null
        +options: []
      }
      5 => verbb\formie\models\IntegrationField {#1542
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "phone"
        +name: "Telefoonnummer"
        +type: "string"
        +required: null
        +options: []
      }
      6 => verbb\formie\models\IntegrationField {#1541
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "email"
        +name: "E-mail"
        +type: "string"
        +required: null
        +options: []
      }
    ]
  }
  3 => verbb\formie\models\IntegrationCollection {#1537
    -_events: []
    -_eventWildcards: []
    -_behaviors: []
    -_errors: null
    -_validators: null
    -_scenario: "default"
    +id: "7894001__78816bff-96de-4684-a943-306c42689a65"
    +name: "[NL] Form"
    +fields: array:2 [
      0 => verbb\formie\models\IntegrationField {#1539
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "trackingID"
        +name: "Tracking ID"
        +type: null
        +required: null
        +options: []
      }
      1 => verbb\formie\models\IntegrationField {#1538
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "email"
        +name: null
        +type: "string"
        +required: null
        +options: []
      }
    ]
  }
  4 => verbb\formie\models\IntegrationCollection {#1534
    -_events: []
    -_eventWildcards: []
    -_behaviors: []
    -_errors: null
    -_validators: null
    -_scenario: "default"
    +id: "7894001__f7233491-a6c4-4fe4-8aa3-96436242dcac"
    +name: "TEMP Newsletter"
    +fields: array:3 [
      0 => verbb\formie\models\IntegrationField {#1536
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "trackingID"
        +name: "Tracking ID"
        +type: null
        +required: null
        +options: []
      }
      1 => verbb\formie\models\IntegrationField {#1533
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "legalConsentOptions"
        +name: "Legal Consent Options"
        +type: null
        +required: null
        +options: []
      }
      2 => verbb\formie\models\IntegrationField {#1535
        -_events: []
        -_eventWildcards: []
        -_behaviors: []
        -_errors: null
        -_validators: null
        -_scenario: "default"
        +handle: "email"
        +name: "Email"
        +type: "string"
        +required: null
        +options: []
      }
    ]
  }
]
```

**Comment by engram-design** [comment]
Thanks for confirming, indeed it's an array of `IntegrationCollection` objects, which is very, very strange. I'm not sure where that's coming from.

And this is when you've filled out the form on the front-end? I can't see how `IntegrationCollection` objects are passed to that function.

Wondering if you can modify that previous line into the following:

```php
if (isset($fieldSettings[0]) && $fieldSettings[0] instanceof \verbb\formie\models\IntegrationCollection) {
    throw new \Exception('HubSpot Error');
}
```

Which will give us some additional information about _where_ this is coming from. If you can send through the stack trace from that error, that should show the full path and lifecycle that it took to get to this point.

**Comment by jodeba** [comment]
Hi,

I'm actually using the dropdown in the Control Panel "Integrations (Select an option) > Hubspot" to trigger the Integration.
I also set the option "Use Queue for Integrations" to false for debugging.

this is the stacktrace:

```
{
  "name": "Exception",
  "message": "API error: “HubSpot Error” C:\\www\\project\\vendor\\verbb\\formie\\src\\base\\Integration.php:565",
  "code": 0,
  "error": "API error: “HubSpot Error” C:\\www\\project\\vendor\\verbb\\formie\\src\\base\\Integration.php:565",
  "exception": "verbb\\formie\\errors\\IntegrationException",
  "file": "C:\\www\\project\\vendor\\verbb\\formie\\src\\base\\Integration.php",
  "line": 136,
  "trace": [
    {
      "file": "C:\\www\\project\\vendor\\verbb\\formie\\src\\integrations\\crm\\HubSpot.php",
      "line": 384,
      "function": "apiError",
      "class": "verbb\\formie\\base\\Integration",
      "type": "::"
    },
    {
      "file": "C:\\www\\project\\vendor\\verbb\\formie\\src\\services\\Submissions.php",
      "line": 283,
      "function": "sendPayload",
      "class": "verbb\\formie\\integrations\\crm\\HubSpot",
      "type": "->"
    },
    {
      "file": "C:\\www\\project\\vendor\\verbb\\formie\\src\\controllers\\SubmissionsController.php",
      "line": 931,
      "function": "sendIntegrationPayload",
      "class": "verbb\\formie\\services\\Submissions",
      "type": "->"
    },
    {
      "function": "actionRunIntegration",
      "class": "verbb\\formie\\controllers\\SubmissionsController",
      "type": "->"
    },
    {
      "file": "C:\\www\\project\\vendor\\yiisoft\\yii2\\base\\InlineAction.php",
      "line": 57,
      "function": "call_user_func_array"
    },
    {
      "file": "C:\\www\\project\\vendor\\yiisoft\\yii2\\base\\Controller.php",
      "line": 178,
      "function": "runWithParams",
      "class": "yii\\base\\InlineAction",
      "type": "->"
    },
    {
      "file": "C:\\www\\project\\vendor\\yiisoft\\yii2\\base\\Module.php",
      "line": 552,
      "function": "runAction",
      "class": "yii\\base\\Controller",
      "type": "->"
    },
    {
      "file": "C:\\www\\project\\vendor\\craftcms\\cms\\src\\web\\Application.php",
      "line": 301,
      "function": "runAction",
      "class": "yii\\base\\Module",
      "type": "->"
    },
    {
      "file": "C:\\www\\project\\vendor\\craftcms\\cms\\src\\web\\Application.php",
      "line": 625,
      "function": "runAction",
      "class": "craft\\web\\Application",
      "type": "->"
    },
    {
      "file": "C:\\www\\project\\vendor\\craftcms\\cms\\src\\web\\Application.php",
      "line": 280,
      "function": "_processActionRequest",
      "class": "craft\\web\\Application",
      "type": "->"
    },
    {
      "file": "C:\\www\\project\\vendor\\yiisoft\\yii2\\base\\Application.php",
      "line": 384,
      "function": "handleRequest",
      "class": "craft\\web\\Application",
      "type": "->"
    },
    {
      "file": "C:\\www\\project\\web\\index.php",
      "line": 12,
      "function": "run",
      "class": "yii\\base\\Application",
      "type": "->"
    }
  ]
}
```

**Comment by engram-design** [comment]
Thanks for that and confirming how you're triggering the integration. I have no idea _how_ that's stored values like that.

Would it be possible for you to send a support request through, so I can see the content of your form integration settings? I feel like something isn't quite right there.

Head to Formie > Settings > Get Support and if you can select the form that you're testing HubSpot with.

**Comment by jodeba** [comment]
Hi,

I send it from my localhost :)

Thank you

**Comment by engram-design** [comment]
Haven't received your support request yet, sorry! Just wanted to make sure that it sent all okay?

Otherwise, you can also go to Formie > Settings > Import/Export and export the form, sending it through to `support@verbb.io`.

**Comment by engram-design** [comment]
Thanks for that, that's certainly helped get to the bottom of the issue. Fixed for the next release. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.21",
  "...": "..."
}
```

Then run `composer update`.

**Comment by jodeba** [comment]
Thank you very much! 🙂

**Comment by engram-design** [comment]
Fixed in [2.0.22](https://github.com/verbb/formie/releases/tag/2.0.22)
